update-agent-loader/tests/functional_tests.rs

```diff
@@ -0,0 +1,124 @@
+use ed25519_dalek::{Signer, SigningKey};
+use tokio::runtime::Runtime;
+use wiremock::{
+    matchers::{method, path},
+    Mock, MockServer, ResponseTemplate,
+};
+
+/// Load the test signing key that was generated by build.rs
+fn load_test_signing_key() -> SigningKey {
+    let manifest_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
+    let secret_key_path = std::path::Path::new(&manifest_dir)
+        .join("target")
+        .join("debug")
+        .join("keys")
+        .join("secret_key.bin");
+
+    // Read the key from disk
+    let secret_key_bytes = std::fs::read(secret_key_path)
+        .expect("Failed to read test secret key. Make sure to run in debug mode.");
+
+    SigningKey::from_bytes(secret_key_bytes.as_slice().try_into().unwrap())
+}
+
+/// Sign a binary with the specified keypair
+/// Returns a new Vec<u8> containing the original binary plus signature data
+fn sign_binary(data: &[u8], signing_key: &SigningKey) -> Vec<u8> {
+    // Create the signature
```

Comment:
Should we just use ed25519ph with the signing context instead of concatenating a signing context?

And the signing context should be a const

Reply:
you can disregard this - a signing context still makes good sense to do, but this code is for magic strings.